applications/RANSApplication/custom_strategies/bossak_scalar_scheme.h

```diff
@@ -0,0 +1,371 @@
+        rBossakConstants.C2 = 1.0 / (rBossakConstants.Gamma * DeltaTime);
+        rBossakConstants.C3 = (1.0 - rBossakConstants.Gamma) / rBossakConstants.Gamma;
+    }
+
+    template <class TItemType>
+    void AddMassMatrixToRHS(
+        TItemType& rItem,
+        LocalSystemVectorType& rRHS_Contribution,
+        const int ThreadId)
+    {
+        rItem.GetSecondDerivativesVector(mSecondDerivativeValuesVector[ThreadId], 0);
+        (mSecondDerivativeValuesVector[ThreadId]) *= (1.00 - mBossak.Alpha);
+        rItem.GetSecondDerivativesVector(mSecondDerivativeValuesVectorOld[ThreadId], 1);
+        noalias(mSecondDerivativeValuesVector[ThreadId]) +=
+            mBossak.Alpha * mSecondDerivativeValuesVectorOld[ThreadId];
+
+        noalias(rRHS_Contribution) -=
+            prod(mMassMatrix[ThreadId], mSecondDerivativeValuesVector[ThreadId]);
+    }
+
+    template <class TItemType>
+    void CalculateDynamicSystem(
+        TItemType& rItem,
+        LocalSystemMatrixType& rLHS_Contribution,
+        LocalSystemVectorType& rRHS_Contribution,
+        typename TItemType::EquationIdVectorType& rEquationIdVector,
+        const ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_TRY;
+
+        BaseType::CalculateSystemContributions(rItem, rLHS_Contribution, rRHS_Contribution,
+                                               rEquationIdVector, rCurrentProcessInfo);
+
+        const int k = OpenMPUtils::ThisThread();
+
+        rItem.CalculateMassMatrix(mMassMatrix[k], rCurrentProcessInfo);
+        // adding mass contribution to the dynamic stiffness
+        if (mMassMatrix[k].size1() != 0) // if M matrix declared
+        {
+            AddMassMatrixToRHS<TItemType>(rItem, rRHS_Contribution, k);
+
+            noalias(rLHS_Contribution) += mBossak.C0 * mMassMatrix[k];
+        }
+
+        KRATOS_CATCH("");
+    }
+
+    template <class TItemType>
+    void CalculateDynamicRHS(
+        TItemType& rItem,
+        LocalSystemVectorType& rRHS_Contribution,
+        typename TItemType::EquationIdVectorType& rEquationIdVector,
+        const ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_TRY;
+
+        const int k = OpenMPUtils::ThisThread();
+
+        this->CalculateDampingSystem(rItem, this->mDampingMatrix[k], rRHS_Contribution,
+                                     rEquationIdVector, rCurrentProcessInfo, k);
+
+        rItem.CalculateMassMatrix(mMassMatrix[k], rCurrentProcessInfo);
+        // adding mass contribution to the dynamic stiffness
+        if (mMassMatrix[k].size1() != 0) // if M matrix declared
+        {
+            AddMassMatrixToRHS(rItem, rRHS_Contribution, k);
+        }
+
+        KRATOS_CATCH("");
+    }
+
+    void UpdateScalarRateVariables(ModelPart& rModelPart)
+    {
+        const int number_of_nodes = rModelPart.NumberOfNodes();
+
+#pragma omp parallel for
+        for (int i_node = 0; i_node < number_of_nodes; ++i_node) {
+            auto& r_node = *(rModelPart.NodesBegin() + i_node);
```

Comment:
You can avoid the dereference operator in here by using the iterator. In any case you will need to change it again when introducing the parallel utils. Up to you.